### ISSUE TITLE
[BUG] Chapter 2 Stratified Sampling

### ISSUE THREAD

**Issue by carrickkv2** [ageron/handson-ml2#461]
In Chapter 2 of the book, we use stratified sampling to make sure the training data matches the real data. 
It seems for the real data some of the labels are missing. Thus, when you ran strat_train_set.head() or strat_test_set.head(), you get an error. The error message is "Passing list-likes to .loc or [] with any missing labels is no longer supported.

The recommended action the error message gives is to .reindex() instead of loc as it seems the library has been changed. 
Here is the line of code where the error occurs. 
```python
from sklearn.model_selection import StratifiedShuffleSplit

split = StratifiedShuffleSplit(n_splits=1, test_size=0.2, random_state=42)
for train_index, test_index in split.split(housing, housing["income_cat"]):
    strat_train_set = housing.loc[train_index]
    strat_test_set = housing.loc[test_index]
```

This link explains the official solution. 
https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#deprecate-loc-reindex-listlike

**Comment by ageron** [comment]
Hi @carrickkv2 ,

Thanks for your feedback. However, I can't reproduce this error, since there's no label missing for me. I just ran the [Colab notebook for this chapter](https://colab.research.google.com/github/ageron/handson-ml2/blob/master/02_end_to_end_machine_learning_project.ipynb), and everything ran smoothly:
![image](https://user-images.githubusercontent.com/76661/130348416-f3a98b6e-a3ed-45c9-ba32-79c18a6a8a4f.png)

Could you please double check that you're using the same data and code as in the Colab notebook?

**Comment by carrickkv2** [comment]
Hi @ageron the only way to reproduce the bug is by writing strat_train_set.head() or strat_test_set.head() in the same cell as the code with loc. Based upon the link I sent you, it seems loc is being phased out gradually.

I do hope you try it out so you can reproduce the error.
